src/main/kotlin/com/ampnet/projectservice/service/impl/ProjectServiceImpl.kt

```diff
@@ -61,8 +66,15 @@ class ProjectServiceImpl(
     override fun getAllProjects(pageable: Pageable): Page<Project> = projectRepository.findAll(pageable)
 
     @Transactional(readOnly = true)
-    override fun getActiveProjects(pageable: Pageable): Page<Project> =
-        projectRepository.findByActive(ZonedDateTime.now(), true, pageable)
+    override fun getActiveProjects(pageable: Pageable): Page<ProjectWithWallet> {
+        val activeProjects = projectRepository.findByActive(ZonedDateTime.now(), true, pageable)
+        val activeWallets = walletService.getWallets(activeProjects.toList().map { it.uuid })
+            .filter { isActivated(it) }.associateBy { it.owner }
+        val projectsWithWallets = activeProjects.toList().mapNotNull {
+            getProjectWithActiveWallet(it, activeWallets)
+        }
+        return PageImpl(projectsWithWallets, pageable, projectsWithWallets.size.toLong())
```

Comment:
Check docs for PageImpl constructor: [Spring docs](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/PageImpl.html#PageImpl-java.util.List-org.springframework.data.domain.Pageable-long-)
```suggestion
         return PageImpl(projectsWithWallets, pageable, activeProjects.totalElements)
```
This will return incorrect values sometimes but we will see later how to handle this.